//
// Copyright 2021 Google Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

#ifndef GTXILIB_OOPCLASSES_PROTOS_TYPEDEFS_H_
#define GTXILIB_OOPCLASSES_PROTOS_TYPEDEFS_H_

#include "enums.pb.h"
#include "gtx.pb.h"
#include "result.pb.h"

typedef gtxilib::oopclasses::protos::Point PointProto;
typedef gtxilib::oopclasses::protos::Size SizeProto;
typedef gtxilib::oopclasses::protos::Rect RectProto;
typedef gtxilib::oopclasses::protos::Color ColorProto;
typedef gtxilib::oopclasses::protos::
    ElementType_ElementTypeEnum ElementTypeProto;
typedef gtxilib::oopclasses::protos::UIElement
    UIElementProto;
typedef gtxilib::oopclasses::protos::DisplayMetrics
    DisplayMetricsProto;
typedef gtxilib::oopclasses::protos::DeviceState
    DeviceStateProto;
typedef gtxilib::oopclasses::protos::
    AccessibilityHierarchy AccessibilityHierarchyProto;
typedef gtxilib::oopclasses::protos::TypedValueProto
    TypedValueProto;
typedef gtxilib::oopclasses::protos::MetadataProto
    MetadataProto;
typedef gtxilib::oopclasses::protos::CheckResultProto
    CheckResultProto;
typedef gtxilib::oopclasses::protos::
    AccessibilityEvaluation AccessibilityEvaluationProto;

#endif  // GTXILIB_OOPCLASSES_PROTOS_TYPEDEFS_H_
